generic3g/OuterMetaComponent/read_restart.F90

```diff
@@ -14,6 +14,31 @@ module subroutine read_restart(this, importState, exportState, clock, unusable,
       class(KE), optional, intent(in) :: unusable
       integer, optional, intent(out) :: rc
 
+      ! Locals
+      type(GriddedComponentDriver), pointer :: driver
+      type(ESMF_GridComp) :: gc
+      character(:), allocatable :: name
+      type(MultiState) :: states
+      type(ESMF_State) :: internal_state, import_state
+      type(ESMF_Geom) :: geom
+      type(Restart) :: rstrt
+      integer :: status
+
+      driver => this%get_user_gc_driver()
+      name = driver%get_name()
+      if ((name /= "cap") .and. (name /= "HIST")) then
```

Comment:
Need generic method for this conditional.   And in any event the logic should be reversed:
``` if (name == 'cap' .or. name == 'HIST') then
         ... return?
      end if
      ! no else...

Reply:
This is temporary solution till we have a better way to identify a gridcomp that reads/writes a restart.
We are not returning at this point, instead recursing over children.

Reply:
I am thinking that this one needs further discussion with @bena-nasa and @atrayano 

Our usual rule-of-thumb is to checkpoint all imports.   But HIST makes a mockery of this decision.   The MAPL2 solution (ignoring the fact that MAPL2 implemented history quite differently), would be for the individual collections to explicitly say not to checkpoint their imports when they advertise them.      The generic layer would then not need to do anything special.

Note that "cap" will generally not have any imports (or internal state) and therefore the check here is actually not needed even now. (If I've understood correctly.)